crates/holochain_zome_types/src/clone.rs

```diff
@@ -0,0 +1,62 @@
+//! Cells can be cloned to create new cells with the different properties.
+
+use crate::cell::{CellId, CloneId};
+use derive_more::Display;
+use holo_hash::DnaHash;
+use holochain_integrity_types::DnaModifiers;
+
+/// The arguments to create a clone of an existing cell.
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+pub struct CreateCloneCellInput {
+    /// The id of the cell to clone.
+    pub cell_id: CellId,
```

Comment:
how about this parameter allows for either role name or cell id, similar to the conductor api calls?

Reply:
I'm not seeing that being offered anywhere on the app or admin interfaces? There's a `CloneCellId` which can be either a `CloneId` (newtype of `String`) or the `CellId`. I'm just wanting to make sure I'm not making another type for something that exists but I do like this idea, I'll add this now

Reply:
Ahh actually, this involves introducing the concept of `InstalledAppId` into `holochain_zome_types` which is something I was trying to avoid. Apps shouldn't need to know how they are installed.

In most cases I expect the clone call will be made against the current cell anyway. I would be up for adding another HDK function that just hides getting the DNA info and current agent public key to construct the current cell id, that might be a nicer interface to not need to have a cell id at all?

Reply:
Ah yeah, then nevermind. We could make the interface even simpler and remove the `cell_id`, so that the clone is always created of the cell you're calling. But it's fine like this too. I think we leave it as it is now.